apis/exoscale/v1/dbaas_types.go

```diff
@@ -0,0 +1,155 @@
+package v1
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	exoscaleoapi "github.com/exoscale/egoscale/v2/oapi"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type DBaaSParameters struct {
+	// TerminationProtection protects against termination and powering off.
+	TerminationProtection bool `json:"terminationProtection,omitempty"`
+	// Version is the (major) version identifier for the instance.
+	Version string `json:"version,omitempty"`
+	// Size contains the service capacity settings.
+	Size SizeSpec `json:"size,omitempty"`
+	// +kubebuilder:validation:Enum=ch-gva-2;ch-dk-2;de-fra-1;de-muc-1;at-vie-1;bg-sof-1
+	// +kubebuilder:validation:Required
+
+	// Zone is the datacenter identifier in which the instance runs in.
+	Zone string `json:"zone"`
+
+	IPFilter IPFilter `json:"ipFilter,omitempty"`
+}
+
+// NodeState describes the state of a service node.
+type NodeState struct {
+	// Name of the service node
+	Name string `json:"name,omitempty"`
+	// Role of this node.
+	Role exoscaleoapi.DbaasNodeStateRole `json:"role,omitempty"`
+	// State of the service node.
+	State exoscaleoapi.DbaasNodeStateState `json:"state,omitempty"`
+}
+
+// Notification contains a service message.
+type Notification struct {
+	// Level of the notification.
+	Level exoscaleoapi.DbaasServiceNotificationLevel `json:"level,omitempty"`
+	// Message contains the notification.
+	Message string `json:"message,omitempty"`
+	// Type of the notification.
+	Type exoscaleoapi.DbaasServiceNotificationType `json:"type,omitempty"`
+	// Metadata contains additional data.
+	Metadata runtime.RawExtension `json:"metadata,omitempty"`
+}
+
+// BackupSpec contains settings to control the backups of an instance.
+type BackupSpec struct {
+	// TimeOfDay for doing daily backups, in UTC.
+	// Format: "hh:mm:ss".
+	TimeOfDay TimeOfDay `json:"timeOfDay,omitempty"`
+}
+
+// MaintenanceSpec contains settings to control the maintenance of an instance.
+type MaintenanceSpec struct {
+	// +kubebuilder:validation:Enum=monday;tuesday;wednesday;thursday;friday;saturday;sunday;never
+
+	// DayOfWeek specifies at which weekday the maintenance is held place.
+	// Allowed values are [monday, tuesday, wednesday, thursday, friday, saturday, sunday, never]
+	DayOfWeek exoscaleoapi.DbaasServiceMaintenanceDow `json:"dayOfWeek,omitempty"`
+
+	// TimeOfDay for installing updates in UTC.
+	// Format: "hh:mm:ss".
+	TimeOfDay TimeOfDay `json:"timeOfDay,omitempty"`
+}
+
+var timeOfDayRegex = regexp.MustCompile("^([0-2]?[0-3]):([0-5][0-9]):([0-5][0-9])$")
```

Comment:
This regex patter does not compile values for hour 4 - 9 (also 04 - 09). Maybe you could try `^(2[0-3]|[01]?[0-9]):([0-5][0-9]):([0-5][0-9])$`.

Reply:
how could I be so dumb :see_no_evil: good catch!